src/treq/test/local_httpbin/parent.py

```diff
@@ -0,0 +1,176 @@
+
+    _process = attr.ib(init=False, default=None)
+    _process_description = attr.ib(init=False, default=None)
+
+    _open = staticmethod(open)
+
+    def _spawn_httpbin_process(self, reactor):
+        """
+        Spawn an ``httpbin`` process, returning a :py:class:`Deferred`
+        that fires with the process transport and result.
+        """
+        server = _HTTPBinServerProcessProtocol(
+            all_data_received=self._all_data_received,
+            terminated=self._terminated
+        )
+
+        argv = [
+            sys.executable,
+            '-m',
+            'treq.test.local_httpbin.child',
+        ]
+
+        if self._https:
+            argv.append('--https')
+
+        with self._open(self._error_log_path, 'wb') as error_log:
+            endpoint = endpoints.ProcessEndpoint(
+                reactor,
+                sys.executable,
+                argv,
+                childFDs={
+                    1: 'r',
+                    2: error_log.fileno(),
+                },
+            )
+            # Processes are spawned synchronously.
+            spawned = endpoint.connect(
+                # ProtocolWrapper, WrappingFactory's protocol, has a
+                # disconnecting attribute.  See
+                # https://twistedmatrix.com/trac/ticket/6606
+                policies.WrappingFactory(
+                    protocol.Factory.forProtocol(lambda: server),
+                ),
+            )
+
+        def wait_for_protocol(connected_protocol):
+            process = connected_protocol.transport
+            return self._all_data_received.addCallback(
+                return_result_and_process, process,
+            )
+
+        def return_result_and_process(description, process):
+            return description, process
+
+        return spawned.addCallback(wait_for_protocol)
+
+    def server_description(self, reactor):
+        """
+        Return a :py:class:`Deferred` that fires with the the process'
+        :py:class:`_HTTPBinDescription`, spawning the process if
+        necessary.
+        """
+        if self._process is None:
+            ready = self._spawn_httpbin_process(reactor)
+
+            def store_and_schedule_termination(description_and_process):
+                description, process = description_and_process
+
+                self._process = process
+                self._process_description = description
+
+                reactor.addSystemEventTrigger("before", "shutdown", self.kill)
+
+                return self._process_description
+
+            return ready.addCallback(store_and_schedule_termination)
+        else:
+            return succeed(self._process_description)
+
+    def kill(self):
+        """
+        Kill the ``httpbin`` process.
+        """
+        if not self._process:
+            return
+
+        if platform.system() == "Windows":
+            signo = signal.SIGTERM
+            self._process.signalProcess("TERMINATE")
```

Comment:
Good question!  It looks like [`KILL` also happens to work](https://github.com/Almad/twisted/blob/master/twisted/internet/_dumbwin32proc.py#L240-L241).  With your suggested change to `\n` from `os.linesep`, I think I can eliminate all platform-specific code.